crates/ironrdp-dvc-pipe-proxy/src/worker.rs

```diff
@@ -132,21 +141,50 @@ async fn process_client<P: OsPipe>(ctx: &mut WorkerCtx) -> Result<NextWorkerStat
     }
 }
 
-async fn worker<P: OsPipe>(mut ctx: WorkerCtx) -> Result<(), DvcPipeProxyError> {
+async fn worker<P: OsPipe>(ctx: WorkerCtx) -> Result<(), DvcPipeProxyError> {
+    // Create a bridge between std::sync::mpsc and tokio for async compatibility.
+    let (async_tx, async_rx) = tokio::sync::mpsc::unbounded_channel();
```

Comment:
suggestion: Add a comment about why it’s fine to use an unbounded channel here (= we are bounded by the to_pipe_rx channel).